pkgs/applications/networking/p2p/retroshare/default.nix

```diff
@@ -4,22 +4,21 @@
 
 stdenv.mkDerivation rec {
   name = "retroshare-${version}";
-  version = "0.6.2";
+  version = "0.6.4";
 
   src = fetchFromGitHub {
     owner = "RetroShare";
     repo = "RetroShare";
     rev = "v${version}";
-    sha256 = "0hly2x87wdvqzzwf3wjzi7092bj8fk4xs6302rkm8gp9bkkmiiw8";
+    sha256 = "14shxjzx3szbqlmil05q3n5pwsfgb42lr82clwkq62zqw7170az1";
+    fetchSubmodules = true;
   };
 
   # NIX_CFLAGS_COMPILE = [ "-I${glib.dev}/include/glib-2.0" "-I${glib.dev}/lib/glib-2.0/include" "-I${libxml2.dev}/include/libxml2" "-I${sqlcipher}/include/sqlcipher" ];
 
-  patchPhase = ''
-    # Fix build error
-    sed -i 's/UpnpString_get_String(es_event->PublisherUrl)/es_event->PublisherUrl/' \
-      libretroshare/src/upnp/UPnPBase.cpp
-  '';
+  patches = [ ./0000-libupnp-fixes.patch
```

Comment:
Please add comments linking to upstream tracking of these patches, or explanations for why nix-specific patches are needed.